File: Grade_2/2-1/DS/0422/hoimoon.py
import sys

arr = sys.stdin.readline().rstrip()
stack = []
mid = len(arr) // 2

for idx, value in enumerate(arr):
    if mid > idx:
        stack.append(value)
    elif (mid == idx) and (len(arr) % 2 == 1):
        continue
    else:
        chk = stack.pop()
        if chk != value:
            print(f'두 문자가 다릅니다. cur: {value} stack: {chk}')
            break

File: Grade_2/2-1/DS/study/0422/hoimoon.c
#include <stdio.h>
#include <string.h>

int main (){
    char arr[100];
    char stack[100];
    int top = -1;
    scanf("%s", arr);

    int length = strlen(arr);
    int mid = length / 2;

    for (int i = 0; i < length; i++){
        if (i < mid){
            top += 1;
            stack[top] = arr[i];
        }
        else if (i == mid && length % 2 == 1) {
            continue;
        }
        else {
            if (stack[top] == arr[i])
                top -= 1;
            else
                printf("두 문자가 다릅니다. %c %c", stack[top], arr[i]);
                break;
        }
    }
}

